Publish github action pipeline fails as required tokens are not set

Seems that called workflows don't have access to the required variables. The caller workflow has to capture them and pass them on to the callee it seems
see
https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations


```
[NPM PUBLISH] Using custom version number for unstable packages: 0.5.1-unstable-20220[62](https://github.com/hyperledger/caliper/runs/7009099279?check_suite_focus=true#step:4:63)2170900
[NPM PUBLISH] Publishing @hyperledger/caliper-core@0.5.1-unstable-20220622170900 with tag "unstable". Attempt 1/5:
NPM_TOKEN must be set when publishing to the public NPM registry.
[NPM PUBLISH] Failed to publish package @hyperledger/caliper-core@0.5.1-unstable-20220622170900 (attempt 1/5)
[NPM PUBLISH] Error: Failed to execute "/home/runner/work/caliper/caliper/packages/caliper-publish/artifacts/npm-publish.sh" with return code 1.
[NPM PUBLISH] Publishing @hyperledger/caliper-core@0.5.1-unstable-20220622170900 with tag "unstable". Attempt 2/5:
NPM_TOKEN must be set when publishing to the public NPM registry.
[NPM PUBLISH] Failed to publish package @hyperledger/caliper-core@0.5.1-unstable-20220622170900 (attempt 2/5)
[NPM PUBLISH] Error: Failed to execute "/home/runner/work/caliper/caliper/packages/caliper-publish/artifacts/npm-publish.sh" with return code 1.
[NPM PUBLISH] Publishing @hyperledger/caliper-core@0.5.1-unstable-20220622170900 with tag "unstable". Attempt 3/5:
NPM_TOKEN must be set when publishing to the public NPM registry.
[NPM PUBLISH] Failed to publish package @hyperledger/caliper-core@0.5.1-unstable-20220622170900 (attempt 3/5)
[NPM PUBLISH] Error: Failed to execute "/home/runner/work/caliper/caliper/packages/caliper-publish/artifacts/npm-publish.sh" with return code 1.
[NPM PUBLISH] Publishing @hyperledger/caliper-core@0.5.1-unstable-20220622170900 with tag "unstable". Attempt 4/5:
NPM_TOKEN must be set when publishing to the public NPM registry.
[NPM PUBLISH] Failed to publish package @hyperledger/caliper-core@0.5.1-unstable-20220622170900 (attempt 4/5)
[NPM PUBLISH] Error: Failed to execute "/home/runner/work/caliper/caliper/packages/caliper-publish/artifacts/npm-publish.sh" with return code 1.
[NPM PUBLISH] Publishing @hyperledger/caliper-core@0.5.1-unstable-20220622170900 with tag "unstable". Attempt 5/5:
NPM_TOKEN must be set when publishing to the public NPM registry.
[NPM PUBLISH] Failed to publish package @hyperledger/caliper-core@0.5.1-unstable-20220622170900 (attempt 5/5)
[NPM PUBLISH] Error: Failed to execute "/home/runner/work/caliper/caliper/packages/caliper-publish/artifacts/npm-publish.sh" with return code 1.
[NPM PUBLISH] Aborting, could not publish package @hyperledger/caliper-core@0.5.1-unstable-20220622170900
Error: Process completed with exit code 1.
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -10,3 +10,4 @@ jobs:
   publish-caliper:
     uses: ./.github/workflows/publish.yml
     needs: unit-tests
+    secrets: inherit

